### PR TITLE
[rewire] Added generics to rewire

### PR DESCRIPTION
**Pull request by CaselIT**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jhnns/rewire
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dslint/dt.json" }`.

Added generics to rewire using the default values introduced in typescript 2.3
The build at the moment fails, but I'm not sure if dtslint support typescript 2.3

### CODE REVIEW

**Comment by CaselIT** [comment]
Waiting for to start supporting ts2.3 in about a month. See https://github.com/Microsoft/dtslint/issues/31 for details

File: types/rewire/index.d.ts [hunk 1]
```diff
@@ -1,36 +1,34 @@
-// Type definitions for rewire v2.5.1
+// Type definitions for rewire 2.5
 // Project: https://github.com/jhnns/rewire
 // Definitions by: Borislav Zhivkov <https://github.com/borislavjivkov>
+//                 Federico Caselli <https://github.com/CaselIT>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Typescript Version: 2.3
```
Comment (andy-ms) [inline types/rewire/index.d.ts]: Change to `TypeScript`.

Comment (CaselIT) [inline types/rewire/index.d.ts]: done

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**